### PR TITLE
Implement long running polling

### PR DESCRIPTION
**Pull request by p3ck**
##### SUMMARY
This implements long running polling the same way the GenericRestClient does with a 600 second overall timeout and 30 second polling.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
plugins/inventory/azure_rm.py

##### ADDITIONAL INFORMATION
I ran the inventory integration tests but I don't have enough hosts in my azure instance to recreate the original issue.  So all I can do is verify that it still works with my limited test environment.

### CODE REVIEW

**Comment by p3ck** [comment]
Possible fix for #1614

**Comment by Fred-sun** [comment]
@p3ck There's no problem on my end. But I don't have enough virtual machines on my side to repeat the problem. So I will push to merge #1614 users after they have no problem synchronizing updates. Thank you!

File: plugins/inventory/azure_rm.py [hunk 1]
```diff
@@ -463,30 +472,61 @@ def _process_queue_batch(self):
                     # FUTURE: error-tolerant operation mode (eg, permissions)
                     # FUTURE: store/handle errors from individual handlers
                     result.handler(r['content'], **result.handler_args)
+                    batch_processed.append(returned_name)
+                elif status_code in _RETRY_CODES:
+                    # 429: Too many requests Error, Backoff and Retry
+                    batch_retry = True
+            if batch_retry:
+                time.sleep(backoff_factor * (2 ** (retry_count)))
```
Comment (xuzhang3) [inline plugins/inventory/azure_rm.py:480]: Regarding the backoff policy, retry time should based on the response. Other wise the retry request will be rejected too.
https://learn.microsoft.com/en-us/troubleshoot/azure/virtual-machines/windows/troubleshooting-throttling-errors#throttling-error-details

**Review by xuzhang3 — CHANGES_REQUESTED** [review]

File: plugins/inventory/azure_rm.py (hunk 1 above)
Comment (nirarg) [reply plugins/inventory/azure_rm.py:480]: Thank you @xuzhang3,
@p3ck is on vacation (he will be back next week) so I'm checking how to add the RetryPolicy class to be used here, in order to have this response retry time to be considered

Comment (nirarg) [reply plugins/inventory/azure_rm.py:480]: The issue here is that in the current implementation the wait sleep is done once for the all batch together
Do you have any suggestion/thoughts how to change it to work for each vm separately?

Comment (xuzhang3) [reply plugins/inventory/azure_rm.py:480]: @nirarg  I see, misunderstand the retry policy here.  The backoff policy and retry is focus on the local batch requests here so we cannot determine the retry time for all the request.

**Review by xuzhang3 — APPROVED** [review]

**Review by nirarg — APPROVED** [review]

File: plugins/inventory/azure_rm.py (hunk 1 above)
Comment (Klaas-) [reply plugins/inventory/azure_rm.py:480]: https://github.com/Azure/azure-sdk-for-python/issues/36285 I think the retry stuff should be implemented on the python sdk level which azure_rm inventory uses. That could be a nice generic solution, but so far I have not convinced MSFT it's a good idea :)

**Pull request merged**